http-server/src/main/java/xtdb/jackson/CallDeserializer.java

```diff
@@ -0,0 +1,36 @@
+package xtdb.jackson;
+
+import clojure.lang.Keyword;
+import clojure.lang.PersistentHashMap;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.node.BaseJsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import xtdb.IllegalArgumentException;
+import xtdb.tx.Call;
+
+import java.io.IOException;
+import java.util.List;
+
+public class CallDeserializer extends StdDeserializer<Call> {
+
+    public CallDeserializer() {
+        super(Call.class);
+    }
+
+    @Override
+    public Call deserialize(JsonParser p, DeserializationContext ctxt) throws IllegalArgumentException, IOException {
+        ObjectMapper mapper = (ObjectMapper) p.getCodec();
+        BaseJsonNode node = mapper.readTree(p);
```

Comment:
it's not idiomatic to rely on `try`/`catch` in Java for casting exceptions (if that's what this and the other cases are trying to do) - best to `if (node instanceof ObjectNode) { ... } else { throw ...}`